### PR TITLE
Basic Reimbursement File

### PR DESCRIPTION
**Pull request by bh2smith**
CoW Protocol Solvers are to be reimbursed on a recurring schedule with their eth spent on successful batch settlements + a reward of 100 COW tokens for each batch settled. This PR provides a basic functional version of the query returning a safe-airdrop compatible transfer file.

As a first PR to this repo we have included 

1. A file writer
2. Simple query to transfer records for an accounting period
3. A script to fetch and write results as a csv transfer file a compatible with `safe-airdrop`

## Test Plan

1. Follow the installation guide in readme to set up the project.
2. Fill out your own dune credentials in `.env`
3. Run the following script

```
python -m src.fetch.transfer_file --start '2022-02-01' --end '2022-02-08'
```
and get the following CSV written to the project's `out` directory

[transfers-2022-02-01-to-2022-02-08.csv](https://github.com/bh2smith/cow-solver-rewards/files/8178108/transfers-2022-02-01-to-2022-02-08.csv)

Alternatively, you can visit the dune page linked in the query introduced here and download the CSV transfer file.

4. Visit the Gnosis Safe interface and upload this file there (or paste it).

### CODE REVIEW

**Comment by MartinquaXD** [comment]
Hmm, I see your point. Since this PR only queries the data and doesn't modify or alter it in any way not having a test for it is probably fine.
But if future PRs introduce some logic to transform data in any way it would be a good thing to commit the data input as well as the output. Then you could have an automated test which simply runs the logic on the stored input and compares the output with what is stored in the repo. If dune does some internal shenanigans which would result in a different input to the system, so be it. But we could still be sure that the logic of the code didn't change during development.

**Comment by bh2smith** [comment]
Thanks again @MartinquaXD for these valuable comments. My plan for testing our ETL will be as follows.

Split up the (E) Extraction from the (T) Transformation. This means that we can test our transformation methods by passing in dummy data.

Similar, but not entirely related. I plan Implement a (non-redundant) form of "lazy loading" similar to the token-allocation project which would allow us to pass in a file instead of fetch. In this project, however it will be a generic wrapper so we dont have a bunch of `if load_from is not None: record_set.load_from;`.


I will circle back and try to fix up the comment you made on skipping write.

File: src/read_write.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    def __str__(self):
+        return self.filename()
+
+
+def write_to_csv(data_list: list, outfile: File):
+    """Writes `data_list` to `filename` as csv"""
+    if len(data_list) == 0:
+        print("No data in list, skipping write")
+        return
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply src/read_write.py]: Great points here! Have removed.

**Review by MartinquaXD — APPROVED** [review]

**Comment by MartinquaXD** [comment]
@bh2smith and I had a call talking a bit about formalizing the idea around `extract` and `transform` a bit more.
Since this might be a bigger post I will create a detailed issue and will link it here when it's done.

**Pull request merged**